tests/test_penalties.py

```diff
@@ -759,3 +763,44 @@ def test_penalty(self, random_ragged_cmf):
 
         with pytest.raises(TypeError):
             penalty.penalty(A)
+
+
+class TestTemporalSmoothness(BaseTestFactorMatricesPenalty):
+    PenaltyType = penalties.TemporalSmoothnessPenalty
+    penalty_default_kwargs = {"smoothness_l": 1}
+
+    n_rows = 10
+    min_rows = n_rows
+    max_rows = n_rows
+
+    def test_penalty(self, random_regular_cmf):
+        # Check that the penalty term is computed correctly.
+
+        penalty = self.PenaltyType(**self.penalty_default_kwargs)
+        cmf, random_ragged_shapes, rank = random_regular_cmf
+        weights, (A, B_is, C) = cmf
+
+        K = len(B_is)
+
+        penalty_term = 0
+        for k in range(1, K):
+            penalty_term += tl.norm(B_is[k] - B_is[k - 1]) ** 2
```

Comment:
This is a minor nit, but if we use `tl.sum` instead of doing the norm, then we'll get fewer roundoff errors and it might be slightly (but probably not measurably) faster as we skip a square root followed by a square

```suggestion
            penalty_term += tl.sum((B_is[k] - B_is[k - 1]) ** 2)
```